ee/app/livechat-enterprise/server/lib/LivechatEnterprise.js

```diff
@@ -161,3 +164,56 @@ export const LivechatEnterprise = {
 		updateRoomPriorityHistory(roomId, user, priority);
 	},
 };
+
+const RACE_TIMEOUT = 1000;
+
+const queueWorker = {
+	running: false,
+	queues: [],
+	async start() {
+		if (this.running) {
+			return;
+		}
+
+		await this.getActiveQueues();
+		await OmnichannelQueue.initQueue();
+		this.running = true;
+		return this.execute();
+	},
+	async stop() {
+		this.running = false;
+		return OmnichannelQueue.stopQueue();
+	},
+	async getActiveQueues() {
+		// undefined = public queue(without department)
+		return [undefined].concat(await LivechatInquiry.getDistinctQueuedDepartments());
```

Comment:
I couldn't think on a better approach yet, but having an `undefined` item is weird 😂

Reply:
definitely 🙃